packages/core/src/analysis/SubstanceAnalysis.ts

```diff
@@ -318,6 +333,8 @@ export const cascadingDelete = <T>(
         return findArg(s, id);
       } else if (s.tag === "Decl") {
         return s.name === id;
+      } else if (s.tag === "LabelDecl") {
+        return s.variable.value === id.value;
```

Comment:
Would it make sense to use `switch` for this instead of `if`/`else`? That would allow us to automatically check exhaustivity.

Reply:
Good catch! Helped me handle a few more cases too.